src/main/java/seedu/duke/Storage.java

```diff
@@ -499,98 +494,293 @@ public void updatePair() {
-                            }
-                        }
+        scanner = new Scanner(pairFile);
+        while (scanner.hasNext()) {
+            String[] pairingParameters = scanner.nextLine().split("\\s\\:\\s");
+            boolean hasSplitCorrectly = pairingParameters.length == 2;
+            if (hasSplitCorrectly) {
+                String[] clientParameters = pairingParameters[0].split("\\s\\|\\s");
+                String[] propertyParameters = pairingParameters[1].split("\\s\\|\\s");
+                boolean hasCorrectClientLength = clientParameters.length == 4;
+                boolean hasCorrectPropertyLength = clientParameters.length == 4;
+                boolean hasCorrectFormat = hasCorrectClientLength && hasCorrectPropertyLength;
 
-                        for (int i = 0; i < propertyList.getCurrentListSize(); i += 1) {
-                            String propertyLandLordName = propertyList.getPropertyList().get(i).getLandlordName();
-                            String propertyListAddress = propertyList.getPropertyList().get(i).getPropertyAddress();
-                            String propertyListRentalPrice = propertyList.getPropertyList().get(i).getRentingPrice();
-                            String propertyListUnitType = propertyList.getPropertyList().get(i).getUnitType();
-                            boolean isCorrectLandlordName = propertyLandLordName.equals(landLordName);
-                            boolean isCorrectAddress = propertyListAddress.equals(propertyAddress);
-                            boolean isCorrectRental = propertyListRentalPrice.equals(rentalPrice);
-                            boolean isCorrectUnitType = propertyListUnitType.equals(unitType);
-
-                            isCorrectProperty = isCorrectLandlordName && isCorrectAddress
-                                    && isCorrectRental && isCorrectUnitType;
-                            if (isCorrectProperty) {
-                                pairingProperty = propertyList.getPropertyList().get(i);
-                            }
-                        }
+                if (hasCorrectFormat) {
+                    try {
+                        Client pairingClient = getPairingClient(clientParameters);
+                        Property pairingProperty = getPairingProperty(propertyParameters);
+
+                        boolean isCorrectClient = pairingClient != null;
+                        boolean isCorrectProperty = pairingProperty != null;
 
                         if (isCorrectProperty && isCorrectClient) {
                             pairingList.addPairing(pairingClient, pairingProperty);
                         }
-
+                    } catch (DukeException e) {
+                        UI.showExceptionMessage(e);
                     }
+                } else {
+                    skipPairingEntries();
                 }
+            } else {
+                skipPairingEntries();
```

Comment:
There seems to be some arrowhead code here

Reply:
Arrowhead code has been fixed in the latest commit. Thanks for identifying :)